doc/xversionmessage.md

```diff
@@ -0,0 +1,250 @@
+frequent synchronization with other implementations and while avoiding
+double-assignments. Of course, regular integration into a common
+directory should take place. The specifics of this and the exact,
+detailed interpretation of the fields in the `xmap` will obviously
+rely on external specifications that cannot ever be the scope of this
+document.
+
+The `xmap` directory is meant to be machine readable. For this reason,
+a simple `python` tool in the Bitcoin Unlimited implementation
+produces a C++ header file from this directory (see below).
+
+The rationale to not directly implement it as a simple C++ header is
+that this will make it easier for non-C++ implementations to extract
+the desired values from this table as well as ensuring that table
+stays simple.
+
+## `xmap` directory format
+The file is assumed to be ASCII-encoded and read
+line-by-line. Anything that follows a `#` is deemed a comment and
+ignored. Lines that are empty or consisting only of whitespace are
+ignored as well.
+
+Each line is interpreted as a string tuple. The entries in the tuple
+are separated by any positive amount of whitespace in the directory file.
+
+String tuples that have `KEY` as their first entry specify the key type, naming,
+prefix, suffix and data type of an entry in the `xmap`, in this order. For example:
+
+`KEY i BU_LISTEN_PORT                            0x0000                   0x0000         u64c`
+
+
+The name of a key should be an all-upper-case, underscore-separated string
+that is a valid C++ `enum` identifier, like in the above example.
+
+The keys in the table should be marked with either `INITIAL` or
+CHANGEABLE, expressed as `i` or `c`. Currently, only the meaning of
+INITIAL (`i`) is defined.
+
+The idea is that an initial value is marking values that are valid for
+the lifetime of the connection to the peer, whereas values marked as
+`c` could be updated with a to-be-defined update mechanism. This
+functionality, however, has not been implemented yet. And implementer
+that desires the functionality of a `CHANGEABLE` key is expected to
+update this very specification accordingly.
+
+Valid values for the data type specification are currently `vector` for an
+unspecified vector of bytes and `u64c` for a value that is assumed to
+be a compact-size encoded 64-bit unsigned integer like described
+above.
+
+Other tuples that do not start with `KEY` or have the above format are
+currently undefined and should be expected to produce an error in any
+`xmap` directory parsers that might be implemented (or potentially in
+any of the later processing and compilation stages).
+
+## Implementation prefixes
+
+Assigning prefixes for known implementations in alphabetical order,
+BitcoinABC should use the `0x0000` prefix, BCash the `0x0001` prefix,
+Bitcoin Unlimited the `0x0002` prefix, BitcoinXT the `0x0003` prefix.
+Experimental or temporary features should use the `0x1000` prefix.
+
+(FIXME: List all known implementations. Feedback welcome)
```

Comment:
I was hoping to avoid collisions and clashes by doing it like I did. I can, however, go and reverse 0xbeef for fields of the DAGURVAL type, if you want? :-)

Reply:
No need, I'll just pick what I want without asking permission 😛

Reply:
Alright, but please let us all know then whether you claimed 0xbeef (or whatever). To avoid chaos.